src/sentry/profiles/java.py

```diff
@@ -112,17 +112,37 @@ def convert_android_methods_to_jvm_frames(methods: list[dict]) -> list[dict]:
     return [
         {
             "function": m["name"],
+            "index": i,
             "module": m["class_name"],
             "signature": m["signature"],
         }
-        for m in methods
+        for i, m in enumerate(methods)
     ]
 
 
 def merge_jvm_frames_with_android_methods(frames: list[dict], methods: list[dict]) -> None:
-    assert len(frames) == len(methods)
-    for f, m in zip(frames, methods):
-        m["class_name"] = f["module"]
-        m["name"] = f["function"]
-        if "signature" in f:
-            m["signature"] = f["signature"]
+    for f in frames:
+        m = methods[f["index"]]
+        if m.get("data", {}).get("deobfuscation_status", "") != "deobfuscated":
+            m["class_name"] = f["module"]
+            m["name"] = f["function"]
+            m["data"] = {"deobfuscation_status": "deobfuscated"}
+            m["inline_frames"] = []
+            if "signature" in f:
+                m["signature"] = f["signature"]
+            if "lineno" in f:
+                m["source_line"] = f["lineno"]
+            if "abs_path" in f:
+                m["source_file"] = f["abs_path"]
+            if "in_app" in f:
+                m["in_app"] = f["in_app"]
```

Comment:
Remind me again, the first frame in the list of deobfuscated frame with the matching index is the non inline frame right? And the ones after were inlined?

Reply:
Current handling of those frames show them `reversed` so I'll iterate from the bottom.